src/chapters/angular-lsn2/studio.rst

```diff
@@ -0,0 +1,265 @@
+       :alt: Animated gif of crew member name being clicked, edited, and then saved.
+
+       Example of crew member name being edited.
+
+We need to add a click event to the member name. To do that, put ``{{member.name}}`` inside of a
+``<span>`` that has a ``(click)`` handler. Make your ``<li>`` in ``src/app/crew/crew.component.html``
+look like the below code.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <li *ngFor="let member of crew">
+      <span (click)="edit(member)" class="editable-text">{{member.name}}</span>
+      <button (click)="remove(member)">remove</button>
+   </li>
+
+Add this property to the crew component. ``memberBeingEdited`` represents the crew
+member that is currently being edited.
+
+.. sourcecode:: TypeScript
+
+   memberBeingEdited: object = null;
+
+Next we need to add the below ``edit`` function to the crew component file ``src/app/crew/crew.component.ts``.
+This function will set a ``memberBeingEdited`` variable to be equal to the crew member that was clicked.
+
+.. sourcecode:: TypeScript
+
+   edit(member: object) {
+      this.memberBeingEdited = member;
+   }
+
+Now we need to add an ``*ngIf`` that will show the two versions of the member, the display state or the edit state.
+In the edit state an input box with a save button will appear, but for now the input and save won't have any functionality.
+Make your ``src/app/crew/crew.component.html`` file look like the below code.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <h3>Crew</h3>
+   <ul>
+      <li *ngFor="let member of crew">
+
+         <span *ngIf="memberBeingEdited !== member; else elseBlock">
+            <!-- display state of member -->
+            <span (click)="edit(member)" class="editable-text">{{member.name}}</span>
+            <span *ngIf="member.firstMission">
+               - 1st
+            </span>
+            <button (click)="remove(member)">remove</button>
+         </span>
+
+         <ng-template #elseBlock>
+            <!-- edit state of member -->
+            <input />
+            <button>save</button>
+         </ng-template>
+
+      </li>
+   </ul>
+   <input #name type="text"/>
+   <label>First mission<input #firstMission type="checkbox"/></label>
+   <button (click)="add(name.value, firstMission.checked)">Add</button>
+
+Finally we are going to make the edit state of the member update the member name when save is clicked.
+Update the ``<input>`` and ``<button>`` tags to look like the below.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <ng-template #elseBlock>
+      <!-- edit state of member -->
+      <input #updatedName (keyup.enter)="save(updatedName.value, member)" [value]="member.name"/>
+      <button (click)="save(updatedName.value, member)">save</button>
+   </ng-template>
+
+The last step is to add the ``save`` function to the crew component. This function will be called
+when the ``<button>`` is clicked or when the enter keys is pressed when the ``<input>`` is focused.
```

Comment:
Typo fix and rephrasing suggestion: "... the enter key is pressed inside the input box."